cfme/physical/provider/__init__.py

```diff
@@ -31,18 +40,45 @@ def __init__(
 
     @variable(alias='db')
     def num_server(self):
-        pass
+        provider = self.appliance.rest_api.collections.providers.find_by(name=self.name)[0]
+        num_server = 0
+        for server in self.appliance.rest_api.collections.physical_servers:
+            if server['ems_id'] == provider.id:
+                num_server += 1
+        return num_server
 
     @num_server.variant('ui')
     def num_server_ui(self):
-        pass
+        try:
+            num = self.get_detail("Relationships", 'Physical Servers')
```

Comment:
get_detail is a convenience method from the old implementations of navigation/web_ui widgets.

It would be better to directly use the view and its SummaryTable widgets for this.

```python
view = navigate_to(self, 'Details')
try:
    num = view.relationships.get_text_of('Physical Servers')
except NameError:
    logger.warning('Failed to fetch Physical Servers detail, trying Nodes')
    num = view.relationships.get_text_of('Nodes')